### PR TITLE
Add HalfNormal distribution

### PR DESCRIPTION
**Pull request by shwinnn**
Adds the HalfNormal distribution. I force it to be bounded below at 0 to match the APIs of other PPL libraries (e.g. [numpyro](https://numpyro.readthedocs.io/en/v0.2.0/distributions.html#halfnormal), [tensorflow probability](https://www.tensorflow.org/probability/api_docs/python/tfp/distributions/HalfNormal)

### CODE REVIEW

File: mcx/distributions/halfnormal.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+from mcx.distributions import constraints
+from mcx.distributions.distribution import Distribution
+
+
+class HalfNormal(Distribution):
+    params_constraints = {
+        "sigma": constraints.strictly_positive,
+    }
+    support = constraints.real
```
Comment (rlouf) [inline mcx/distributions/halfnormal.py:12]:
```suggestion
    support = constraints.positive
```

**Review by rlouf — CHANGES_REQUESTED** [review]
Thank you for taking the time to add this distribution! I added a few comments. Before merging I would need a few tests, which you can probably get from modifying the Normal distribution's test. And also a test that makes sure that values out of support return `-inf`. Thank you again.

**Comment by rlouf** [comment]
I would really like to merge this before the first release. Is there anything I can do to help?

**Comment by shwinnn** [comment]
Hi, sorry I just haven't found the time to write the tests. If I have the downtime I will have a look but I don't have a specific timeframe for you I'm afraid!

**Comment by rlouf** [comment]
No problem, take your time!

**Comment by rlouf** [comment]
It looks good to me, merging. Thank you for taking the time to add the `HalfNormal` distribution to MCX!

**Comment by rlouf** [comment]
Actually, would you mind squashing all your commits? Github doesn't suggest you as a co-author if I squash the commits myself here, and I wouldn't want you to not appear as a contributor.

**Review by rlouf — CHANGES_REQUESTED** [review]
Just squash your commits and we're good to go.

**Comment by rlouf** [comment]
Merging, thanks again!

**Pull request merged**